app/Http/Controllers/AppealController.php

```diff
@@ -545,4 +550,40 @@ public function privacyhandle(Request $request,$id,$action) {
         else {abort(401);}
         return view('appeals.publicappeal', ['id'=>$id,'info' => $info, 'comments' => $logs, 'userlist'=>$userlist, 'replies'=>$replies]);
     }
+
+    public function showVerifyOwnershipForm(Request $request, Appeal $appeal, $token)
+    {
+        abort_if($appeal->verify_token !== $token, 400, 'Invalid token');
+        return view('appeals.verifyaccount', ['appeal' => $appeal]);
+    }
+
+    public function verifyAccountOwnership(Request $request, Appeal $appeal)
+    {
+        $request->validate([
+            'verify_token' => ['required', new SecretEqualsRule($appeal->verify_token)],
+            'secret_key' => ['required', new SecretEqualsRule($appeal->appealsecretkey)]
+        ]);
+
+        $appeal->update([
+            'verify_token' => null,
+            'user_verified' => true,
+        ]);
+
+        $ua = $request->server('HTTP_USER_AGENT');
+        $ip = $request->server('HTTP_X_FORWARDED_FOR');
+        $lang = $request->server('HTTP_ACCEPT_LANGUAGE');
+
+        Log::create([
+            'user' => 0,
+            'referenceobject'=> $appeal->id,
+            'objecttype'=>'appeal',
+            'action'=>'account verification',
+            'reason' => 'account ownership verified',
+            'ip' => $ip,
+            'ua' => $ua . " " .$lang
+        ]);
+
+        // todo: use post?
+        return redirect()->to('/publicappeal?hash=' . $appeal->appealsecretkey);
```

Comment:
This one I'm not really sure what to do with, I'd like to not include the secret key in URLs but you can't really redirect to a post URL

Reply:
Fixed by #52 right? So you could code this pending approval of that (which is close)

Reply:
Yep